screeps-game-api/src/constants.rs

```diff
@@ -176,7 +176,7 @@ pub mod find {
 
 enum_from_primitive! {
     #[repr(i32)]
-    #[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
+    #[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Deserialize)]
```

Comment:
Unless I'm mistaken this will deserialize via strings, not numbers. If it works, great.

If not, this will probably need a custom implementation which calls one of the methods to get from an i32.

Reply:
The problem I'm thinking of is that direction constants actually are numbers in JS, not strings (like some of the other constants are).

See https://github.com/screeps/common/blob/f0214b373843fbe0e1d750108a5b8a1ebb34f1a9/lib/constants.js#L44 - they are defined as their integer numbers.